iceoryx_hoofs/source/posix_wrapper/timer.cpp

```diff
@@ -113,11 +112,7 @@ void Timer::OsTimer::callbackHelper(sigval data) noexcept
             // prohibits other threads from entering the flag protected region
             handle.m_callbackIsAboutToBeExecuted.test_and_set(std::memory_order_acq_rel);
 
-            if (handle.m_timer == nullptr)
-            {
-                errorHandler(Error::kPOSIX_TIMER__INCONSISTENT_STATE);
-                return;
-            }
+            cxx::Expects((handle.m_timer != nullptr) && "Timer in inconsistent state");
```

Comment:
Please use the error handler here and in all remaining files where you added an `cxx::Expects` / `cxx::Ensures` or where you removed the `errorHandler`.

Reply:
Rejected, quote from `doc/design/error-handling.md`:

`The error handler cannot be used in hoofs.`

See [this comment](https://github.com/eclipse-iceoryx/iceoryx/pull/1100#discussion_r807013547) for the background an my opinion. For `v2.0` I would like to implement our current error handling strategy.

I'm happy to discuss changes to the error handling strategy in #1032 for `v3.0`.